circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/CopierFactoryManager.java

```diff
@@ -55,14 +57,26 @@ void postConstruct() {
     }
   }
 
-  CopierFactory getCopierFactory(Path sourceLocation, Path replicaLocation) {
+  CopierFactory getCopierFactory(Path sourceLocation, Path replicaLocation, Map<String, Object> copierOptions) {
     String sourceScheme = sourceLocation.toUri().getScheme();
     String replicaScheme = replicaLocation.toUri().getScheme();
-    for (CopierFactory copierFactory : copierFactories) {
-      if (copierFactory.supportsSchemes(sourceScheme, replicaScheme)) {
-        LOG.debug("Found CopierFactory '{}' for sourceScheme '{}' and replicaScheme '{}'",
-            copierFactory.getClass().getName(), sourceScheme, replicaScheme);
-        return copierFactory;
+    if (copierOptions.containsKey(COPIER_FACTORY_CLASS)) {
+      for (CopierFactory copierFactory : copierFactories) {
+        final String copierFactoryClassName = copierFactory.getClass().getName();
+        if (copierFactoryClassName.equalsIgnoreCase(copierOptions.get(COPIER_FACTORY_CLASS).toString())) {
+          LOG.debug("Found CopierFactory '{}' for sourceScheme '{}' and replicaScheme '{}'",
```

Comment:
This message should be changed to indicate that it found the copier using the config and not via the schemes.
```
LOG.debug("Found CopierFactory '{}'  configured by {}",	
+                  copierFactoryClassName, COPIER_FACTORY_CLASS);
```